### PR TITLE
Create docker compose configuration

### LINKED ISSUE #246: create docker compose configuration for cache layer
We need a simple docker compose file to create a containerized service involving Node.js and MongoDB. It should also use a bind mount for development purposes.

### PR DESCRIPTION
**Pull request by doogyb**
Creates a docker network between a Node.js runtime and a Mongodb
database using a docker-compose configuration file.

closes #246

### CODE REVIEW

File: be/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# dapsboard cache layer
+
+To run, first cd inside this directory (the `be` dir) and run
+
+```
+docker compose up
+```
+
+This will expose the server on port `3000`.
```
Comment (mindrones) [inline be/README.md]: guessing this is supposed to be `4000`?

File: be/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      - 4000:4000 
+    environment:
+      PORT: 4000
+  mongo:
+    image: "mongo"
+    volumes:
+      - mongodb:/data/db
+
+volumes:
+  mongodb:
```
Comment (mindrones) [inline be/docker-compose.yml:19]: seems incomplete, leftover?

File: be/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "author": "Samuel Doogan",
+  "bugs": {
+    "url": "https://github.com/nestauk/dapsboard/issues"
+  },
+  "description": "A cache for the ElasticSearch aggregation requests performed by dapsboard",
+  "homepage": "https://github.com/nestauk/dapsboard#readme",
+  "license": "ISC",
```
Comment (mindrones) [inline be/package.json]: not needed as we have a license in the root

File: be/package.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "bugs": {
+    "url": "https://github.com/nestauk/dapsboard/issues"
+  },
+  "description": "A cache for the ElasticSearch aggregation requests performed by dapsboard",
+  "homepage": "https://github.com/nestauk/dapsboard#readme",
+  "license": "ISC",
+  "main": "index.js",
```
Comment (mindrones) [inline be/package.json]: we don't havve this at the moment (if we add it it should be `"main": "src/index.js",`)

File: be/package.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  "name": "dapsboard_cache_layer",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/nestauk/dapsboard.git"
+  },
+  "scripts": {
+    "dev": "nodemon src/server.js"
+  },
+  "type": "module",
+  "version": "1.0.0",
```
Comment (mindrones) [inline be/package.json]: we're using a single `version` in the root, so this can be removed I think

**Review by mindrones — CHANGES_REQUESTED** [review]
first pass

File: be/docker-compose.yml (hunk 2 above)
Comment (doogyb) [reply be/docker-compose.yml:19]: I believe this is needed, it instructs docker compose to create a docker volume if it does not already exist.

File: be/package.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	"husky": {
+		"hooks": {
+			"pre-push": "npm run lint"
+		}
+	},
+	"name": "dapsboard_cache_layer",
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/nestauk/dapsboard.git"
+	},
```
Comment (mindrones) [inline be/package.json]: (1) not needed here :)

File: be/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	"bugs": {
+		"url": "https://github.com/nestauk/dapsboard/issues"
+	},
+	"dependencies": {
+		"husky": "^4.3.0",
+		"mongodb": "^4.13.0",
+		"nodemon": "^2.0.20"
+	},
+	"description": "A cache for the ElasticSearch aggregation requests performed by dapsboard",
+	"homepage": "https://github.com/nestauk/dapsboard#readme",
```
Comment (mindrones) [inline be/package.json]: see (1)

File: be/package.json [hunk 8]
```diff
@@ -0,0 +1,27 @@
+{
+	"bugs": {
+		"url": "https://github.com/nestauk/dapsboard/issues"
+	},
```
Comment (mindrones) [inline be/package.json]: See (1)

**Review by mindrones — COMMENTED** [review]
minor comments re `be/package.json`

**Review by mindrones — APPROVED** [review]
GTM (and welcome! :)

**Pull request merged**